src/htdocs/controller/main_controller.php

```diff
@@ -13,6 +13,21 @@ class MainController extends AbstractController {
 
     private $devices;
 
+    private $supportedAvgTypes = array('0', '0.25', '1', '24');
+
+    private function obtainCurrentAvgType($cookieName) {
+        $currentAvgType = '1'; // default
+        $cookieGetName = str_replace('.', '_', $cookieName); // just PHP things... ;)
```

Comment:
I don't think I fully follow the flow here. So we need to access cookie with a different name than it's set (`$cookieGetName` vs `$cookieName`)? Why don't we pass the name with underscore here and use it everywhere?

Reply:
Well this code is a result of some iteration. Firstly I had cookie names in `camelCase` and everything worked, then I switched to `.` separator with prefixes and for some reason it stopped working.

The reason was that PHP automatically [change the name of the cookie](https://www.php.net/manual/en/reserved.variables.cookies.php#119381) disallowing to get it by the real (and [fully legal](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#attributes)) name.

I liked the dot separator as it is more readable for me and later it can e.g. be used to clear the saved preferences from single subpage (checking it against the prefix). So I decided to stick to the workaround for PHP weirdness inside PHP and not adjust the rest of the world to it (note the cookies in JS).

If you insist I can change it though. 😉

Reply:
I didn't know it, that's strange indeed. Let's use underscores everwhere, so we won't get into this trap elsewhere.

Reply:
I have decided to use `|` as a separator. Should be working fine in PHP. I hope that you are fine with this.